app/background.js

```diff
@@ -41,6 +41,10 @@ const checkToClose = function (cutOff: ?number) {
           tabs = tabs.filter((tab) => tab.pinned === false);
           // Filter out audible tabs if the option to do so is checked
           tabs = tabs.filter((tab) => (tab.audible && settings.get("filterAudio")) === false);
+          // Filter out tabs that are in a group if the option to do so is checked
+          tabs = tabs.filter(
+            (tab) => (tab.groupId > 0 && settings.get("filterGroupedTabs")) === false
+          );
```

Comment:
This can skip the `filter` all together if the setting is disabled. I see the line above could be written that way too.

```suggestion
          tabs = settings.get("filterGroupedTabs") ? tabs.filter((tab) => tab.groupId !== 0) : tabs;
```